crates/apollo-compiler/src/diagnostic.rs

```diff
@@ -171,9 +171,29 @@ impl<T: ToCliReport> ToCliReport for &T {
 type AriadneSpan = (FileId, Range<usize>);
 
 /// Translate a SourceSpan into an ariadne span type.
-fn to_span(location: SourceSpan) -> Option<AriadneSpan> {
-    let start = location.offset();
-    let end = location.end_offset();
+///
+/// When source text is available (looked up from the source map), this function
+/// ensures byte offsets fall on valid UTF-8 character boundaries. This is necessary
+/// because the parser may produce spans that end inside multibyte characters when
+/// there are lexer errors (e.g., unexpected CJK characters in identifiers).
+fn to_span(location: SourceSpan, source_text: Option<&str>) -> Option<AriadneSpan> {
```

Comment:
It's suspicious that `SourceSpan` contains bytes that don't align with character boundaries at this point. Perhaps the root cause is actually inside apollo-parser, it shouldn't produce such spans to begin with...